docs/PROVISIONING_ARCH.md

```diff
@@ -0,0 +1,77 @@
+
+
+```yaml
+---
+# Provision Openstack Instances
+- import_playbook: provision-instances.yml <1>
+
+# Pre-Install Steps
+- import_playbook: pre-install.yml <2>
+
+# Install - this runs openshift-ansible
+- import_playbook: install.yml <3>
+
+# Post Install
+- import_playbook: post-install.yml <4>
+```
+
+1. _Provision Infrastructure_ creates the servers/instances and any other supporting infrastructure requires (network, DNS, load balancers, etc.). In general this phase
+  - Begins with having credentials with proper access to an Infrastructure provider.
+  - Ends with servers provisioned and running with an OS installed, SSH keys (with root or sudo) pushed, and `subscription-manager` installed.
+2. _Pre-Install_ handles all of the host prep you'd expect to happen when preparing to install OpenShift. This phase includes:
+  - Subscribing the hosts via `subscription-manager` (supports individual account or Satellite subscription using _activation keys_)
+  - Installing/configuring the container runtime, including container storage devices
+  - Installing prerequisite packages
+3. _Install_ runs the normal openshift installation playbook using [openshift-ansible](https://github.com/openshift/openshift-ansible.git). This means that any features of the native openshift installer are available to use with CASL provisioning. (see [OpenShift Ansible Integration](#openshift-ansible-integration) below)
+4. _Post Install_ covers any additional configurations or deployments that we would like to have happen that isn't provided by the native installer. This can include:
+  - Syncing additional administrator SSH keys (typically a cloud provider only provisions a single key)
+  - Creating a set of demo users (when using the `HTPassswordIdentityProvider`)
+  - Creation of any sort of OpenShift resources using the [openshift-applier](https://github.com/redhat-cop/openshift-applier.git) automation framework.
+
+Some principles we try to adhere to with all of these phases include the following.
+
+- **Modularity**. Each phase should be runnable on its own, using the exact same inventory
+- **Idempotentcy**. All phases should be safe to re-run at any time without impact.
+- **Commonality**. This is the idea that we should essentially treat all Infrastructure providers the same, allowing us to treat different providers as _pluggable_ to a [common provisioning framework](#pluggable-provisioners).
+
+## OpenShift Ansible integration
+
+CASL-Ansible is intended to be a compliment (and in some cases an upstream) to the [openshift-ansible](https://github.com/openshift/openshift-ansible.git) repo. As such, we intend for the CASL tools to support all of the features and capabilities of the openshift-ansible cluster deployment playbook. The aim is to use CASL to provision and prepare infrastructure and then hand off to the [deploy_cluster](https://github.com/openshift/openshift-ansible/blob/master/playbooks/deploy_cluster.yml) playbook for the installation of OpenShift.
```

Comment:
Maybe start this with "CASL-Ansible **provisioning** is intended ...." as CASL includes a lot more than just provisioning now.